text.py

```diff
@@ -147,8 +147,7 @@ def query(self, query_text, n=10):
         qwords = [w for w in words(query_text) if w not in self.stopwords]
         shortest = argmin(qwords, lambda w: len(self.index[w]))
         docs = self.index[shortest]
-        results = [(sum([self.score(w, d) for w in qwords]), d) for d in docs]
-        results.sort()
+        results = sorted([(sum([self.score(w, d) for w in qwords]), d) for d in docs])
```

Comment:
Your change is an improvement, I agree, but I would probably do it like this:

    def total_score(doc): return sum(self.score(w, doc) for w in words)
    return heapq.nlargest((total_score(doc), doc) for doc in docs)